src/sendmail/mail_sender.py

```diff
@@ -0,0 +1,58 @@
+import os
+import mailtrap as mt
+from jinja2 import Environment, FileSystemLoader, select_autoescape
+env = Environment(
+    loader=FileSystemLoader("src/sendmail"),
+    autoescape=select_autoescape()
+)
+
+def send_email(args):
+    MAILTRAP_API_TOKEN = os.environ.get("MAILTRAP_API_TOKEN")
+    MAILTRAP_SENDER_ADDRESS = os.environ.get("MAILTRAP_SENDER_ADDRESS") 
+    MAILTRAP_TO_ADDRESS = os.environ.get("MAILTRAP_TO_ADDRESS")
+    MAILTRAP_BCC_ADDRESS = os.environ.get("MAILTRAP_BCC_ADDRESS")
+
+   
+    mainTemplate = env.get_template("template.html")
+    dataTemplate = env.get_template("record.html")
+
+    templates = []
+    
+    # For each record, fill the data template, then put the data into the main template
+    for record in args:    
+        templates.append(dataTemplate.render(
+        record_number=record.record_number,
+        record_type=record.record_type.value,
+        project_name=record.project_name,
+        address=record.address,
+        expiration_date=record.expiration_date,
+        short_notes=record.short_notes,
+        record_link=record.record_details_link
+    ))
```

Comment:
Can we pass the `BuildingServicesSearchResult` object itself into `render` rather than decomposing it into its constituent fields?

Reply:
If you're having trouble w/ a cyclic import, you can fix it by extracting `schema.py`:

```
from dataclasses import dataclass
from enum import Enum
from typing import Any


class PermitType(str, Enum):
    commercial_wrecking_permit = "Building/Wrecking/Commercial/NA"
    residential_wrecking_permit = "Building/Wrecking/Residential/NA"


@dataclass
class BuildingServicesSearchResult:
    # TODO validation (regex?)
    record_number: str  # e.g. WRK2024R-00138
    record_details_link: str  # URL to the record details page
    record_type: PermitType
    project_name: str | None
    address: str
    expiration_date: Any | None  # I haven't found any records w/ this yet
    short_notes: str | None

    def __str__(self):
        return f"{self.record_type.value}:\n{self.record_number} - {self.project_name} - {self.address}"

```

and then `from src.schema import BuildingServicesSearchResult`

Reply:
Got this working locally. 

```
def send_email(records: list[BuildingServicesSearchResult]):
    MAILTRAP_API_TOKEN = os.environ.get("MAILTRAP_API_TOKEN")
    MAILTRAP_SENDER_ADDRESS = os.environ.get("MAILTRAP_SENDER_ADDRESS")
    MAILTRAP_TO_ADDRESS = os.environ.get("MAILTRAP_TO_ADDRESS")
    MAILTRAP_BCC_ADDRESS = os.environ.get("MAILTRAP_BCC_ADDRESS")

    mainTemplate = env.get_template("template.html")
    dataTemplate = env.get_template("record.html")

    filled_template = mainTemplate.render(
        records=[dataTemplate.render(record=record) for record in records]
    )
```
You'll just need to update `record.html` to use object notation:

```
<tr>
    {% if record.record_number %}
        <td>{{ record.record_number }}</td>
    {% else %}
        <td></td>
    {% endif %}
    # etc
```